command/harness/setup.go

```diff
@@ -97,30 +94,43 @@ func HandleSetup(ctx context.Context, r *SetupVMRequest, s store.StageOwnerStore
 		return nil, fmt.Errorf("pool not defined")
 	}
 
+	cleanUpFn := func() {
+		errCleanUpStageOwner := s.Delete(context.Background(), stageRuntimeID)
+		if errCleanUpStageOwner != nil {
+			logr.WithError(errCleanUpStageOwner).Errorln("failed to delete failed stage owner")
+		}
+	}
+
+	if err := s.Create(ctx, &types.StageOwner{StageID: stageRuntimeID, PoolName: pool}); err != nil {
+		return nil, fmt.Errorf("could not create stage owner entity: %w", err)
+	}
+
 	instance, err := poolManager.Provision(ctx, pool, env.Runner.Name, env.Settings.LiteEnginePath)
 	if err != nil {
+		go cleanUpFn()
 		return nil, fmt.Errorf("failed provisioning")
```

Comment:
Let's add the error in the return here. I saw "failed provisioning" in the logs but it doesn't print the error

Reply:
Done